p2p/discovery_legacy/mdns.go

```diff
@@ -0,0 +1,215 @@
+package discovery_legacy
+
+import (
+	"context"
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p/p2p/discovery"
+
+	logging "github.com/ipfs/go-log/v2"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+	"github.com/whyrusleeping/mdns"
```

Comment:
So... how hard would it be to use the _new_ zeroconf package with the _old_ protocol? The old package has a lot of race conditions, etc.

Reply:
We could, but I really don't want to. This is a legacy package anyway, and I don't want to inadvertently break something here.

Reply:
My concern is that it has some safety issues. But yeah, that's probably fine. You're right about it being legacy.

(maybe document this)